src/shared/components/challenge-listing/index.jsx

```diff
@@ -22,6 +22,29 @@ import './style.scss';
 const CHALLENGE_PLACEHOLDER_COUNT = 8;
 
 export default function ChallengeListing(props) {
+  function CheckDateTime(str) {
+    const reg = /^(\d{4})-(\d{1,2})-(\d{1,2})T(\d{1,2}):(\d{1,2}):(\d{1,2})\.(\d{3})Z$/;
+    const r = str.match(reg);
+    if (r == null) return false;
+    r[2] -= 1;
+    const d = new Date(r[1], r[2], r[3], r[4], r[5], r[6]);
+    if (d.getFullYear() !== r[1]) return false;
+    if (d.getMonth() !== r[2]) return false;
+    if (d.getDate() !== r[3]) return false;
+    if (d.getHours() !== r[4]) return false;
+    if (d.getMinutes() !== r[5]) return false;
+    if (d.getSeconds() !== r[6]) return false;
+    return true;
+  }
+  const filterState = props.filterState;
```

Comment:
@twicoder And, although it probably will do the trick, the best way to make the check and avoid using the date in filter, if it is invalid, is inside reducer (the filter query params are picked up from URL during server-side rendering, thus checking it there will guarantee that invalid dates won't be used anywhere): https://github.com/topcoder-platform/community-app/blob/develop/src/shared/reducers/challenge-listing/index.js#L272

Also, once at the client-side, we should remove invalid dates from query. This can be done at this point. To update query params use this utility function: https://github.com/topcoder-platform/community-app/blob/develop/src/shared/utils/url.js#L20